src/driver/cockroachdb/CockroachDriver.ts

```diff
@@ -461,16 +461,14 @@ export class CockroachDriver implements Driver {
 
                 // manually convert enum array to array of values (pg does not support, see https://github.com/brianc/node-pg-types/issues/56)
                 value = (value as string)
-                    .substr(1, (value as string).length - 2)
+                    .slice(1, -1)
                     .split(",")
                     .map((val) => {
                         // replace double quotes from the beginning and from the end
                         if (val.startsWith(`"`) && val.endsWith(`"`))
                             val = val.slice(1, -1)
-                        // replace double escaped backslash to single escaped e.g. \\\\ -> \\
-                        val = val.replace(/(\\\\)/g, "\\")
-                        // replace escaped double quotes to non-escaped e.g. \"asd\" -> "asd"
-                        return val.replace(/(\\")/g, '"')
+                        // replace escaped backslash and double quotes
+                        return val.replace(/\\(\\|")/g, '$1')
```

Comment:
Do we have unit tests around this that we can use to validate the behavior we want?

This will escape differently compared to the prior code. What should `\\\\"` produce? The old version results in `"` and this version results in `\\"`

Reply:
I don't have time to setup CRDB for this, but the test that could be updated is `test/functional/database-schema/enums-array/*`.

CodeQL detected this correctly, double escaping should not happen, everything should be escaped at once.

Reply:
@mguida22 I added some tests for the escape characters, but they would pass anyway. The reason is that this function is used when reading the value, and the value is already escaped correctly by the CRDB server/`pg` library.

Basically, this would produce incorrect results only if we would get an incorrectly escaped value, which we don't.

Nevertheless, I'd still do this fix, to avoid any issues that might arise if the code gets copy-pasted and to make CodeQL stop complaining.